### PR TITLE
Add additional costs field in `EthereumTransactionBody`

### LINKED ISSUE #254: Add a new field in `EthereumTransactionBody` for additional non-gas related costs
### Description
With the addition of hollow accounts by HIP-583, we want relayers to be able to specify the allowance they are willing to give for the costs of hollow account creation. Due to security reasons, the currently existing `max_gas_allowance` field cannot be used for this purpose.

### Solution
Add a new field for additional non-gas costs allowance in `EthereumTransactionBody` definition.

### PR DESCRIPTION
**Pull request by dimitar-dinev**
**Description**:

Adds a `max_additional_costs_allowance` field to `EthereumTransactionBody`.

**Related issue(s)**:

Fixes #254

### CODE REVIEW

File: services/ethereum_transaction.proto [hunk 1]
```diff
@@ -60,4 +60,12 @@ message EthereumTransactionBody {
    * the entire fee.
    */
   int64 max_gas_allowance = 3;
+
+  /**
+   * The maximum amount, in tinybars, that the payer of the Hedera transaction
+   * is willing to pay to for any additional *non-gas* costs of the transaction.
```
[the remark this answers is not in the document]
Comment (dimitar-dinev) [reply services/ethereum_transaction.proto]: Fixed

File: services/ethereum_transaction.proto [hunk 2]
```diff
@@ -60,4 +60,12 @@ message EthereumTransactionBody {
...
+   *
+   * An example of such costs would be any hollow account creation fees.
+   */
+  int64 max_additional_costs_allowance = 4;
```
[the remark this answers is not in the document]
Comment (dimitar-dinev) [reply services/ethereum_transaction.proto]: Yup,  `max_additional_fee_allowance` is clearer IMO, changed now

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM !

**Review by david-bakin-sl — APPROVED** [review]
LGTM

**Review by Nana-EC — CHANGES_REQUESTED** [review]
Blocking as we may not need this any longer.

**Comment by dimitar-dinev** [comment]
Closing this as a different fee-charging strategy involving multiple gas prices was chosen, and these changes are no longer needed.

**Pull request closed without merging**